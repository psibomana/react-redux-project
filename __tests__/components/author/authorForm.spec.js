import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { createMemoryHistory } from 'history'
import AuthorForm from '../../../src/components/author/authorForm'

describe('Author Form Test', () => {
  let wrapper, author, mockOnChange, mockOnSave, mockOnDelete, history;

  beforeEach(() => {
    author = {
      id: "firstName-lastName",
      firstName: "firstName",
      lastName: "lastName"
    }

    mockOnChange = jest.fn();
    mockOnSave = jest.fn();
    mockOnDelete = jest.fn();
    history = createMemoryHistory('/authors')
    wrapper = shallow(
      <AuthorForm
        author={author}
        onChange={mockOnChange}
        onSave={mockOnSave}
        onDelete={mockOnDelete}
        history={history}
        />
    )
  })

  it('should render as expected', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('should call onChange handler from first name', () => {

    const event = {
      target: { value: 'updated-firstname' }
    };

    wrapper.find('#firstName').simulate('change', event);
    expect(mockOnChange.mock.calls.length).toEqual(1);
  });

  it('should call onChange handler from last name', () => {

    const event = {
      target: { value: 'updated-lastname' }
    };

    wrapper.find('#lastName').simulate('change', event);
    expect(mockOnChange.mock.calls.length).toEqual(1);
  });
})
