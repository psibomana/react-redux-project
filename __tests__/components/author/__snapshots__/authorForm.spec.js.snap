// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Form Test should render as expected 1`] = `
<form>
  <h1>
    Manage Author
  </h1>
  <TextInput
    id="firstName"
    label="First Name"
    name="firstName"
    onChange={[Function]}
    value="firstName"
  />
  <TextInput
    id="lastName"
    label="Last Name"
    name="lastName"
    onChange={[Function]}
    value="lastName"
  />
  <input
    className="btn btn-primary"
    id="saveBtn"
    onClick={[Function]}
    type="button"
    value="Save"
  />
  <input
    className="btn btn-danger"
    id="deleteBtn"
    onClick={[Function]}
    type="button"
    value="Delete"
  />
</form>
`;
